pytest-embedded-idf/pytest_embedded_idf/serial.py

```diff
@@ -266,3 +266,32 @@ def is_target_flashed_same_elf(self) -> bool:
             elf_sha256.update(fr.read())
 
         return flash_elf_sha256 == elf_sha256.digest()
+
+
+class LinuxSerial:
+    """
+    IDF Linux serial Dut class
+
+    """
+
+    def __init__(
+        self,
+        app: IdfApp,
+        msg_queue: MessageQueue,
+        target: Optional[str] = None,
+    ) -> None:
+        self.app = app
+        self.msg_queue = msg_queue
+        self.port = None
+
+        if not hasattr(self.app, 'target'):
+            raise ValueError(f'Idf app not parsable. Please check if it\'s valid: {self.app.binary_path}')
+
+        if target and self.app.target and self.app.target != target:
+            raise ValueError(f'Targets do not match. App target: {self.app.target}, Cmd target: {target}.')
+
+        self.start_duplicate_stdout_popen()
+
+    def start_duplicate_stdout_popen(self):
+        os.chmod(self.app.elf_file, 755)
```

Comment:
should be executable already.

Reply:
I am not sure why, but when this file is copied to the temp folder it has `read-only` permission which is causing `PermissionError: [Errno 13] Permission denied:`

Reply:
well, a few more questions.

1. when does this file would be copied to the temp folder? seems like we could use them directly. If you're addressing it while testing, I recommend to use `shutil.copytree` ([ref](https://docs.python.org/3/library/shutil.html#shutil.copytree)) since it would keep the permissions recursively.
2. could  you help check the permission of the elf file you committed? couldn't see it via the github UI.

Reply:
Yes, the file that I committed has execute permissions:
```
-rwxr-xr-x
```